core/components/link.js

```diff
@@ -111,6 +113,32 @@ function pluginsFactory() {
   };
 }
 
+// TODO: Get a full tld list from IANA, or make it an option.
+// scheme :: name :: tld
+export const URL_REGEX =
+  /(^|\s)(((http|https|ftp):\/\/)?(?:[^\s.:\/]+\.)+(?:com|net|io))\s$/;
+
+function autoLinkInputRule(type) {
+  return new InputRule(URL_REGEX, (state, match, start, end) => {
+    if (!match[0]) {
+      return null;
+    }
+    const [_, leadingSpace, text, scheme] = match;
+    // If no scheme, use default scheme http://
+    const href = scheme ? text : `http://${text}`;
```

Comment:
Should we make it default to https?

Reply:
There are quite some website that are http only (e.g. corporate-internal websites). And for most https sites they would redirect http requests to https anyway.